blockchain/validate.go

```diff
@@ -537,10 +537,10 @@ func standaloneToChainRuleError(err error) error {
 	// Convert standalone package rule errors to blockchain rule errors.
 	var rErr standalone.RuleError
 	if errors.As(err, &rErr) {
-		switch rErr.Err {
-		case standalone.ErrUnexpectedDifficulty:
+		switch {
```

Comment:
It seems odd to unwrap this and then use `errors.Is` which it already handles that.  Probably should be:

```Go
// Convert standalone package rule errors to blockchain rule errors.
switch{
case errors.Is(err, standalone.ErrUnexpectedDifficulty):
	return ruleError(ErrUnexpectedDifficulty, err.Error())
...
}

return err
```